This patch has been submitted upstream, and accepted; it should no longer be
needed with the next Python version upgrade:

http://bugs.python.org/issue25004

--- Python-3.4.3/Lib/test/test_mmap.py.~1~	2015-08-28 12:40:08.000000000 -0700
+++ Python-3.4.3/Lib/test/test_mmap.py	2015-09-04 14:51:47.034993552 -0700
@@ -727,7 +727,10 @@
             f.write(tail)
             f.flush()
         except (OSError, OverflowError):
-            f.close()
+            try:
+                f.close()
+            except (OSError, OverflowError):
+                pass
             raise unittest.SkipTest("filesystem does not have largefile support")
         return f
 
